### ISSUE TITLE
Multiple Sessions crashes WinAppDriver

### ISSUE THREAD

**Issue by jeremeguenther** [Microsoft/WinAppDriver#602]
I have an odd issue I am trying to resolve.

I am building out a test suite using Selenium.  Due to Internet Explorer crashing periodically, I kill the browser and restart it after each test.

After each re-start I re-hook the WinAppDriver back into the IE process.  Which works fine, but after about 10 test (haven't counted, but it is always the same number), WinAppDriver throws a 404 error when trying to execute a click command against the browser.

I have a few other ideas to try, but does anyone know if there is some sort of session limit that WinAppDriver has and how to clear out unused sessions?

**Comment by hassanuz** [comment]
> I have a few other ideas to try, but does anyone know if there is some sort of session limit that WinAppDriver has and how to clear out unused sessions?

Can you clarify what you mean by "rehooking" WinAppDriver - do you mean create a new session? There is no arbitrary session limit.

Do you mind also pasting the WinAppDriver.exe log? 

Thank you

**Comment by jeremeguenther** [comment]
Yes by "re-hook" I mean I create a new session in the already running WinAppDriver process.

I have started Closing and Disposing of the WinAppDriver process each time I kill the browser, and that seems to have solved the issue.  Although it definitely feels like a hack.

Does the WinAppDriver.exe write out a log somewhere?  Or are you talking about what gets displayed in the command window while it is running?  If the later, those logs go away whenever the process is closed down, so I do not have them currently.

**Comment by naeemakram** [comment]
The same thing happened to me while I was automating a web application in IE11 and Coded UI. Turns out closing the browser and starting it again after every test is a good idea. Otherwise scripts would keep failing because of an unresponsive web-page. Maybe its the same error sneaked in to WinAppDriver.
